File: backend/models/sensorReading.js
const mongoose = require("mongoose");

const sensorReadingSchema = new mongoose.Schema({
  sensorId: String,
  temperature: Number,
  humidityPercent: Number,
  dateTime: {
    type: Date,
    required: true,
  },
});

const readingSchema = new mongoose.Schema({
  dateTime: {
    type: Date,
    required: true,
    unique: true,
  },
  readings: [sensorReadingSchema],
});

readingSchema.index({ dateTime: 1 });

const SensorReadings = mongoose.model("SensorReadings", readingSchema);

module.exports = SensorReadings;

File: unity_scene/Assets/Scripts/Data/Models/SensorReading.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[System.Serializable]
public class SensorReading
{
    public string dateTime;
    public Reading readings;
}

[System.Serializable]
public class Reading
{
    public string sensorId;
    public float temperature;
    public int humidityPercent;
}